File: modules/core/src/encoder/DateEncoder.ts
import {
  Encoder,
  IntEncoder,
} from '.';

export class DateEncoder implements Encoder<Date> {

  public encode(value: Date, numBits: number): string {

    const intEncoder: IntEncoder = new IntEncoder();

    return intEncoder.encode(Math.round(value.getTime()/100), numBits);

  }

  public decode(value: string): Date {

    const date: Date = new Date();
    const intEncoder: IntEncoder = new IntEncoder();

    date.setTime(intEncoder.decode(value) * 100);

    return date;

  }

}

File: GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/field/DateEncoder.java
package com.gdpr.phase2.encoder.field;

import com.gdpr.phase2.encoder.BaseEncoder;

import java.util.*;

public class DateEncoder implements BaseEncoder<Date> {
//    public String encode(Date value,int numBits) {
//        return IntEncoder.encode(Math.round(value.getTime()/100),numBits);
//    }

    private DateEncoder() {
    }

    private static final DateEncoder instance = new DateEncoder();
    public static DateEncoder getInstance() {
        return instance;
    }
    public Date decode(String value) {
        Date date = new Date(Long.parseLong(value,2)*100);
        return date;
    }
}
